relentless/simulate/lammps.py

```diff
@@ -0,0 +1,315 @@
+
+        # process all lammps commands
+        cmds = ['neighbor {skin} multi'.format(skin=self.neighbor_buffer)]
+        cmds += ['pair_style table linear {N}'.format(N=Nr)]
+        for i,j in sim.ensemble.pairs:
+            # get lammps type indexes, lowest type first
+            id_i,id_j = pair_map(sim,(i,j))
+            cmds += ['pair_coeff {id_i} {id_j} {filename} TABLE_{id_i}_{id_j}'.format(id_i=id_i,id_j=id_j,filename=file_)]
+
+        return cmds
+
+class InitializeFromFile(Initialize):
+    def __init__(self, filename, neighbor_buffer, units='lj', atom_style='atomic'):
+        super().__init__(neighbor_buffer, units, atom_style)
+        self.filename = filename
+
+    def to_commands(self, sim):
+        cmds = super().to_commands(sim)
+        cmds += ['read_data {filename}'.format(filename=self.filename)]
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class InitializeRandomly(Initialize):
+    def __init__(self, neighbor_buffer, seed, units='lj', atom_style='atomic'):
+        super().__init__(neighbor_buffer, units, atom_style)
+        self.seed = seed
+
+    def to_commands(self, sim):
+        cmds = super().to_commands(sim)
+
+        # make box from ensemble
+        box = self.extract_box_params(sim)
+        if not np.all(np.isclose(box[-3:],0)):
+            cmds += ['region box prism {} {} {} {} {} {} {} {} {}'.format(*box)]
+        else:
+            cmds += ['region box block {} {} {} {} {} {}'.format(*box[:-3])]
+        cmds += ['create_box {N} box'.format(N=len(sim.ensemble.types))]
+
+        # use lammps random initialization routines
+        for i in sim.ensemble.types:
+            cmds += ['create_atoms {typeid} random {N} {seed} box'.format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i]-1)]
+        cmds += ['mass * 1.0',
+                 'velocity all create {temp} {seed}'.format(temp=sim.ensemble.T,
+                                                            seed=self.seed)]
+
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class MinimizeEnergy(LAMMPSOperation):
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations, dt):
+        self.energy_tolerance = energy_tolerance
+        self.force_tolerance = force_tolerance
+        self.max_iterations = max_iterations
+
+    def to_commands(self, sim):
+        cmds = ['minimize {etol} {ftol} {maxiter} {maxeval}'.format(etol=self.energy_tolerance,
+                                                                    ftol=self.force_tolerance,
+                                                                    maxiter=self.max_iterations,
+                                                                    maxeval=self.max_iterations)]
```

Comment:
This will allow up to 100 evaluations of the energy per iteration, which is similar (but not exactly the same) as what the HOOMD minimizer does.
```suggestion
                                                                    maxeval=100*self.max_iterations)]
```